marklogic-data-hub-central/ui/src/pages/Run.tsx

```diff
@@ -317,19 +338,25 @@ const Run = (props) => {
                         return axios.get('/api/jobs/' + jobId);
                     }, pollConfig.interval)
                     .then(function(response: any) {
+                        let entityName;
```

Comment:
If the purpose of having targetEntityType on the Job document is to construct a link for exploring search results for a particular entity name, it'd be much easier for the backend to provide that in the initial RunFlowResponse that is returned by running a step. That avoids having to persist additional data, particularly since we'd only be persisting it for a very narrow UI concern. 

In addition, because a Job document may encompass multiple steps, and each step may be associated with a different entity type, we can't have a single "targetEntityType" on a Job document, because we may need multiple values.

But the RunFlowResponse returned by running a step has a Map of step responses, so we can have a targetEntityType on each step response (if one exists).

We also have multiple scenarios to account for here - 1) running a mapping step with a legacy mapping (can't happen in HC, but we need to account for it); 2) running a mapping step via HC; 3) running a matching/merging/mastering step that currently uses "targetEntity" instead of "targetEntityType". Regardless of the approach we take, we need to account for all of these scenarios so that users aren't confused when targetEntityType is only set some of the time.

I recommend opening up a backend task for this, as we'll need to properly define it for the scenarios listed above.

Reply:
@rjrudin This was done by me as a backend task and the commits have since been squashed. I can adjust to account for targetEntity as well. I had thought that was normalized at this point via the function we have on the backend for constructing the flows to run. 

As for the issue of multiple target entities, is that something that we need to accommodate for the running of a single job?

Reply:
I feel more comfortable providing targetEntityType on a per-step basis as part of the API for running a step, as opposed to persisting it in a Job document where it's a lot more difficult to change. 

It looks like that this can be easily implemented/tested too by modifying this method in RunStepResponse:

```
public RunStepResponse withStep(String stepNum) {
```